compiler/rustc_lint/src/context/diagnostics.rs

```diff
@@ -187,6 +187,14 @@ pub(super) fn builtin(
         BuiltinLintDiagnostics::UnexpectedCfgName((name, name_span), value) => {
             let possibilities: Vec<Symbol> =
                 sess.parse_sess.check_config.expecteds.keys().copied().collect();
+
+            #[allow(rustc::potential_query_instability)]
+            let single_possibility =
+                sess.parse_sess.check_config.expecteds.iter().find_map(|(k, v)| match v {
+                    ExpectedValues::Some(v) if v.contains(&Some(name)) => Some(k),
+                    _ => None,
+                });
```

Comment:
We should collect all the possibilities and then sort.
```suggestion
            let mut names_possibilities: Vec<_> =
                sess.parse_sess.check_config.expecteds.iter().filter_map(|(k, v)| match v {
                    ExpectedValues::Some(v) if v.contains(&Some(name)) => Some(k),
                    _ => None,
                }).collect();
            names_possibility.sort();
```